.github/workflows/release-ts.yml

```diff
@@ -0,0 +1,57 @@
+name: Release TypeScript project
+
+on:
+  push:
+    tags:
+      - "*"
+
+permissions:
+  contents: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Extract project name from tag
+        id: extract-project
+        run: |
+          TAG_NAME="${GITHUB_REF#refs/tags/}"
+          PROJECT_NAME="$(echo "$TAG_NAME" | rev | cut -d'-' -f2- | rev)"
+
+          if [[ "$PROJECT_NAME" == *"circuit-lib.js"* ]]; then
+            SUBDIR="circuit-lib/circuit-lib.js"
+          elif [[ "$PROJECT_NAME" == *"cli"* ]]; then
+            SUBDIR="cli"
+          elif [[ "$PROJECT_NAME" == *"compressed-token"* ]]; then
+            SUBDIR="js/compressed-token"
+          elif [[ "$PROJECT_NAME" == *"prover.js"* ]]; then
+            SUBDIR="prover.js"
+          elif [[ "$PROJECT_NAME" == *"stateless.js"* ]]; then
```

Comment:
we need `hasher.rs` too - it's currently a dependency in `stateless.js` and will later become a devDependency

Reply:
does the order of execution matter? 

fwiw, dependency tree looks something like this:
```
rust binaries
circuit-lib.js ->  prover.js
                   hasher.rs  -> stateless.js -> compressed-token -> cli
```

Reply:
does the build and release workflow automate it for all crates/npm packages in one go if so specified, or will we have to open a new release workflow for each? automating this would be tremendously helpful and invite less manual error. the `bump-versions-and-publish-npm.sh` does this too for npm albeit not in ci

Reply:
Currently it's manual for each package and there is no dependency resolution or handling multiple projects at once.

I can try to solve that today. Also, for automatic dependency resolution, it probably makes sense to implement version bumps with semver instead of specifying manually.

Reply:
OK, I changed it now, you can release all TS or all Rust crates at once. Order should be preserved correctly inside one language. It does it in the same way as your script.

Releasing all Rust crates will have to be done first.